itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java

```diff
@@ -1430,6 +1430,84 @@ public Boolean apply(@Nullable NotificationEvent entry) {
             .run(" drop database if exists " + replicatedDbName_CM + " cascade");
   }
 
+  private void injectableForTaskCreation(String primarydb, String replicadb, String tbl, String eventType,
+                                             String eventTypeExpect, WarehouseInstance.Tuple tuple) throws Throwable {
+    List<String> withConfigs = Arrays.asList("'hive.repl.move.optimized.scheme'= ' hDfs , '");
+
+    // fail add notification for given event type.
+    BehaviourInjection<NotificationEvent, Boolean> callerVerifier
+            = new BehaviourInjection<NotificationEvent, Boolean>() {
+      @Nullable
+      @Override
+      public Boolean apply(@Nullable NotificationEvent entry) {
+        if (entry.getEventType().equalsIgnoreCase(eventType) && entry.getTableName().equalsIgnoreCase(tbl)) {
+          injectionPathCalled = true;
+          LOG.warn("Verifier - DB: " + String.valueOf(entry.getDbName())
+                  + " Table: " + String.valueOf(entry.getTableName())
+                  + " Event: " + String.valueOf(entry.getEventType()));
+          return false;
+        }
+        if (entry.getEventType().equalsIgnoreCase(eventTypeExpect) && entry.getTableName().equalsIgnoreCase(tbl)) {
+          nonInjectedPathCalled = true;
+        }
+        return true;
+      }
+    };
+
+    InjectableBehaviourObjectStore.setAddNotificationModifier(callerVerifier);
+    try {
+      replica.load(replicadb, tuple.dumpLocation, withConfigs);
+    } finally {
+      InjectableBehaviourObjectStore.resetAddNotificationModifier();
+    }
+
+    callerVerifier.assertInjectionsPerformed(false, true);
+  }
+
+  @Test
+  public void testTaskCreationOptimization() throws Throwable {
```

Comment:
This test case checks only if INSERT event skips add partition in target during repl load.
But doesn't check if MoveTask is skipped in the DAG. Can we add an unit test for this by stubbing ReplLoadTask? 
Also, need to test this for ACID table too which doesn't seems to skip move task.

Reply:
check if we can add a validation for task presence

Reply:
added test for task creation check